app/src/lib/design-system/components/RQNetworkTable/GenericNetworkTable/components/NetworkPayload.tsx

```diff
@@ -1,41 +1,70 @@
-import React, { useMemo } from "react";
+import React, { useEffect, useMemo, useState } from "react";
 
 import { ThemeProvider } from "@devtools-ds/themes";
 import { ObjectInspector } from "@devtools-ds/object-inspector";
 import { NetworkLogProperty } from "./NetworkLogProperty";
 
 type Props = {
   label?: string;
-  payload: any;
+  payload?: any;
   isPayloadTooLarge?: boolean;
+  fetchPayload?: () => Promise<any>;
```

Comment:
`fetchPayload` this should not be exposed from `NetworkPayload`, ideally `NetworkPayload` should just take the payload, it should not know how to fetch any payload.